projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -506,14 +508,12 @@ private void OnChannelShutdown(ShutdownEventArgs reason)
             _flowControlBlock.Set();
         }
 
-        // TODO async
-        private void OnSessionShutdown(object sender, ShutdownEventArgs reason)
+        private Task OnSessionShutdownAsync(object sender, ShutdownEventArgs reason)
         {
             ConsumerDispatcher.Quiesce();
             SetCloseReason(reason);
             OnChannelShutdown(reason);
-            // TODO async
-            ConsumerDispatcher.Shutdown(reason);
+            return ConsumerDispatcher.ShutdownAsync(reason);
```

Comment:
missiing .ConfigureAwait(false);

Reply:
Thanks, I have no idea why the `CA2007` check didn't catch this.

Reply:
Oh, I'm not actually awaiting the `Task` here, just returning it.